danybeltran/nextjs-typescript-and-mongodb
# Validate Post Form with Zod and Implement Server-Side Data Fetching

### Description

This pull request addresses two main features:

1. **Form Validation with Zod**: 
   - Implemented Zod for form validation in the PostForm component using React Hook Form.
   - Added validation rules to ensure data integrity and user input accuracy.

2. **Server-Side Data Fetching on Posts Page**:
   - Modified the `Posts` page to fetch data from the server.
   - Utilized asynchronous functions to retrieve and display posts from the server.


### How to Test
1. Navigate to the Posts page and observe the server-side data fetching.
2. Visit the crate post page and test form validation with Zod.


### Checklist

- [x] Form validation using Zod implemented.
- [x] Server-side data fetching on the Posts page.
- [x] Code is documented and follows project coding guidelines.
- [x] Tests (if applicable) have been updated or added.
